ui/src/app/pages/SchemaMapping/SchemaMapping.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2020 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react'
+import { SchemaCard } from './SchemaCard'
+import { Services } from 'src/services'
+import { SchemaEmptyState } from './EmptyState'
+import {
+  PureComponent,
+  PureComponentProps,
+  PureComponentState,
+} from '../../components/baseComponent'
+
+/**
+ * Properties
+ */
+
+export interface SchemaMappingProps extends PureComponentProps {
+  artifactId: string
+}
+
+/**
+ * State
+ */
+export interface SchemaMappingState extends PureComponentState {
+  hasKeySchema: boolean
+  hasValueSchema: boolean
+}
+
+export class SchemaMapping extends PureComponent<
+  SchemaMappingProps,
+  SchemaMappingState
+> {
+  constructor(props: Readonly<SchemaMappingProps>) {
+    super(props)
+  }
+
+  public render(): React.ReactElement {
+    return this.state.hasKeySchema || this.state.hasValueSchema ? (
+      <SchemaCard
+        hasKeySchema={this.state.hasKeySchema}
+        hasValueSchema={this.state.hasValueSchema}
+        artifactName={this.props.artifactId}
+      />
+    ) : (
+      <SchemaEmptyState artifactName={this.props.artifactId} />
+    )
+  }
+
+  protected initializeState(): SchemaMappingState {
+    return {
+      hasKeySchema: false,
+      hasValueSchema: false,
+    }
+  }
+
+  // @ts-ignore
+  protected createLoaders(): Promise[] | null {
+    Services.getLoggerService().info(
+      'Loading data for artifact: ',
+      this.props.artifactId,
+    )
+    return [
+      Services.getGroupsService()
+        .getArtifactMetaData('null', this.props.artifactId + '-key', 'latest')
```

Comment:
Don't add hard code params values in api call. Values should be come through props.
Try to make it generic so that it can reuse if needed.

Reply:
Done